src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -345,17 +349,205 @@ bool Pane::ResizePane(const ResizeDirection& direction)
     // either.
     if ((!_firstChild->_IsLeaf()) && _firstChild->_HasFocusedChild())
     {
-        return _firstChild->ResizePane(direction) || _Resize(direction);
+        return _firstChild->ResizePane(direction, amount) || _Resize(direction, amount);
     }
 
     if ((!_secondChild->_IsLeaf()) && _secondChild->_HasFocusedChild())
     {
-        return _secondChild->ResizePane(direction) || _Resize(direction);
+        return _secondChild->ResizePane(direction, amount) || _Resize(direction, amount);
     }
 
     return false;
 }
 
+// Handler for the _root's ManipulationStarted event. We use this to check if a
+// manipulation (read: drag) started inside our content. If it did, we _don't_
+// want to do our normal pane dragging.
+//
+// Consider the case that the TermControl might be selecting text, and the user
+// drags the mouse over the pane border. We don't want that to start moving the
+// border!
+void Pane::_ManipulationStartedHandler(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                                       const winrt::Windows::UI::Xaml::Input::ManipulationStartedRoutedEventArgs& args)
+{
+    // This is added to each _root. But it also bubbles, so only leaves should actually try to handle this.
+    if (args.Handled())
+    {
+        return;
+    }
+    args.Handled(true);
+
+    assert(_IsLeaf());
+
+    const auto contentSize = _content.GetRoot().ActualSize();
+    auto transformCurrentPos = args.Position();
+    auto transformOrigin = transformCurrentPos;
+
+    const auto transform_contentFromOurRoot = _root.TransformToVisual(_content.GetRoot());
+    const auto transformInControlSpace = transform_contentFromOurRoot.TransformPoint(transformOrigin);
+    if ((transformInControlSpace.X > 0 && transformInControlSpace.X < contentSize.x) &&
+        (transformInControlSpace.Y > 0 && transformInControlSpace.Y < contentSize.y))
+    {
+        // clicked on control. bail, and don't allow any manipulations for this one.
+        _shouldManipulate = false;
+    }
+    else
+    {
+        _shouldManipulate = true;

```

Comment:
I think it would be a lot neater if you assigned the member directly here instead of using an if condition.